### ISSUE TITLE
PDep jobs fail due to Species 'collisionModel' AttributeError

### ISSUE THREAD

**Issue by alongd** [ReactionMechanismGenerator/RMG-Py#1220]
Running the ethane-oxidation example on current master (an example with PDep) gives the following trace:

```
After model enlargement:
    The model core has 6 species and 0 reactions
    The model edge has 0 species and 0 reactions


Species with SMILES of [H] named H based on thermo library name
Species with SMILES of [CH2] named CH2(S) based on thermo library name
Species with SMILES of C named CH4 based on thermo library name
Generating kinetics for new reactions...
For reaction CH2(S)(6) + CH4(7) <=> ethane(1) Ea raised from -1.5 to 0 kJ/mol.
Updating 1 modified unimolecular reaction networks (out of 1)...
Updating PDepNetwork #1
Warning: For CC, more characteristic frequencies were generated than vibrational modes allowed. Removed 1 groups (6 frequencies) to compensate.
Warning: Maximum number of iterations reached when fitting spectral data for CC.
Warning: Statmech node 'C_R1' and all its parents have data = None
Warning: For C[CH2], more characteristic frequencies were generated than vibrational modes allowed. Removed 1 groups (2 frequencies) to compensate.
Warning: Statmech node 'C_R0' and all its parents have data = None
Warning: Statmech node 'C_R0' and all its parents have data = None
Traceback (most recent call last):
  File "/home/alongd/Code/RMG-Py//rmg.py", line 173, in <module>
    main()
  File "/home/alongd/Code/RMG-Py//rmg.py", line 167, in main
    rmg.execute(**kwargs)
  File "/home/alongd/Code/RMG-Py/rmgpy/rmg/main.py", line 567, in execute
    bimolecularReact=self.bimolecularReact)
  File "/home/alongd/Code/RMG-Py/rmgpy/rmg/model.py", line 683, in enlarge
    self.updateUnimolecularReactionNetworks()
  File "/home/alongd/Code/RMG-Py/rmgpy/rmg/model.py", line 1642, in updateUnimolecularReactionNetworks
    network.update(self, self.pressureDependence)
  File "/home/alongd/Code/RMG-Py/rmgpy/rmg/pdep.py", line 545, in update
    spec.collisionModel = SingleExponentialDown(alpha0=(4.86,'kcal/mol'))
AttributeError: 'rmgpy.species.Species' object has no attribute 'collisionModel'
```

**Comment by mjohnson541** [comment]
I suspect this is a result of pulling in #1190.

**Comment by alongd** [comment]
Looks like the latest commits from the one_species branch are the cause (although they didn't delete any `collisionModel` attribute).
Checking out 2470e49 and remaking solved this.

**Comment by mliu49** [comment]
My current conclusions:
- The line that breaks is [this one in `rmgpy.rmg.pdep`](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/rmgpy/rmg/pdep.py#L545).
- The cause is removal of the `rmgpy.rmg.model.Species` class.
- The reason that would break this is because `rmgpy.species` is cythonized, whereas `rmgpy.rmg.model` is not. The key difference is that attributes need to be predefined for a cython module but not for a python module.
- The `Species` class has not had a `collisionModel` attribute for a long time (appears to have been replaced by `energyTransferModel` in 4a152a45be5acf32bb9eeb2cd9da7949a8f8825a, 5 years ago).
- Because model species were previously created using the `rmgpy.rmg.model.Species` class, the problematic line was able to create a *new* attribute called collisionModel. Of course, this new attribute is never used.
- At this point in the algorithm, the species already have a value set for `energyTransferModel`.

Remaining questions:
- Since `collisionModel` is never used, could we just get rid of that line?
- Where is `energyTransferModel` being set currently, and is it ok as is?

**Comment by alongd** [comment]
During an RMG run with PDep, CanTherm currently adds to its network input files for bath gases the line:
```
collisionModel = TransportData(shapeIndex=1, epsilon=(810.913,'J/mol'), sigma=(3.621,'angstroms'), dipoleMoment=(0,'C*m'), polarizability=(1.76,'angstroms^3'), rotrelaxcollnum=4.0, comment="""GRI-Mech"""),
```
Perhaps the `collisionModel` above isn't being processed, which is something to look at, but removing the problematic line that creates this attribute in Species might break CanTherm(?)

**Comment by mliu49** [comment]
As far as I can tell, the `collisionModel` in CanTherm input files is completely independent. For one thing, it seems to be a `TransportData` object instead of something like `SingleExponentialDown`. Also, when reading/writting a CanTherm input file, the `collisionModel` parameter is mapped to/from the `Species().transportData` attribute. The naming choices here could have definitely used some improvement...

**Comment by goldmanm** [comment]
The species' `energyTransferModel` parameter is set in `model.makeNewSpecies`, using the `species.generateEnergyTransferModel` function. It does this for all models (even those without pressure dependence). Does anyone know if this was intended for all models? I guess this is a symptom of having one class do all types of RMG models.

Either way, it seems like the creation of this extra parameter can be removed.

**Comment by mliu49** [comment]
Fixed by #1224.
